posts/2019-02-28-Rust-1.33.0.md

```diff
@@ -0,0 +1,158 @@
+
+### Library stabilizations
+
+Here's all of the stuff that's been made `const`:
+
+- [The methods `overflowing_{add, sub, mul, shl, shr}` are now `const`
+  functions for all numeric types.][57566]
+- [The methods `rotate_left`, `rotate_right`, and `wrapping_{add, sub, mul, shl, shr}`
+  are now `const` functions for all numeric types.][57105]
+- [The methods `is_positive` and `is_negative` are now `const` functions for
+  all signed numeric types.][57105]
+- [The `get` method for all `NonZero` types is now `const`.][57167]
+- [The methods `count_ones`, `count_zeros`, `leading_zeros`, `trailing_zeros`,
+  `swap_bytes`, `from_be`, `from_le`, `to_be`, `to_le` are now `const` for all
+  numeric types.][57234]
+- [`Ipv4Addr::new` is now a `const` function][57234]
+
+[57566]: https://github.com/rust-lang/rust/pull/57566
+[57105]: https://github.com/rust-lang/rust/pull/57105
+[57105]: https://github.com/rust-lang/rust/pull/57105
+[57167]: https://github.com/rust-lang/rust/pull/57167
+[57234]: https://github.com/rust-lang/rust/pull/57234
+[57234]: https://github.com/rust-lang/rust/pull/57234
+
+Additionally, these APIs have become stable:
+
+- [`unix::FileExt::read_exact_at`] and [`unix::FileExt::write_all_at`]
+- [`Option::transpose`] and [`Result::transpose`]
+- [`convert::identity`]
+- [`pin::Pin`] and [`marker::Unpin`] (mentioned above)
+- [`marker::PhantomPinned`]
+- [`Vec::resize_with`] and [`VecDeque::resize_with`]
+- [`Duration::as_millis`], [`Duration::as_micros`], and [`Duration::as_nanos`]
+
+[`unix::FileExt::read_exact_at`]: https://doc.rust-lang.org/std/os/unix/fs/trait.FileExt.html#method.read_exact_at
+[`unix::FileExt::write_all_at`]: https://doc.rust-lang.org/std/os/unix/fs/trait.FileExt.html#method.write_all_at
+[`Option::transpose`]: https://doc.rust-lang.org/std/option/enum.Option.html#method.transpose
+[`Result::transpose`]: https://doc.rust-lang.org/std/result/enum.Result.html#method.transpose
+[`convert::identity`]: https://doc.rust-lang.org/std/convert/fn.identity.html
+[`pin::Pin`]: https://doc.rust-lang.org/std/pin/struct.Pin.html
+[`marker::Unpin`]: https://doc.rust-lang.org/stable/std/marker/trait.Unpin.html
+[`marker::PhantomPinned`]: https://doc.rust-lang.org/nightly/std/marker/struct.PhantomPinned.html
+[`Vec::resize_with`]: https://doc.rust-lang.org/std/vec/struct.Vec.html#method.resize_with
+[`VecDeque::resize_with`]: https://doc.rust-lang.org/std/collections/struct.VecDeque.html#method.resize_with
+[`Duration::as_millis`]: https://doc.rust-lang.org/std/time/struct.Duration.html#method.as_millis
+[`Duration::as_micros`]: https://doc.rust-lang.org/std/time/struct.Duration.html#method.as_micros
+[`Duration::as_nanos`]: https://doc.rust-lang.org/std/time/struct.Duration.html#method.as_nanos
+
+See the [detailed release notes][notes] for more details.
+
+### Cargo features
+
+[Cargo should now rebuild a crate if a file was modified during the initial
+build.](https://github.com/rust-lang/cargo/pull/6484/)
+
```

Comment:
@steveklabnik here's some suggested text for the verified email requirement, feel free to edit as you see fit! ❤️ 

```suggestion
### Crates.io 

[As previously announced](https://users.rust-lang.org/t/a-verified-email-address-will-be-required-to-publish-to-crates-io-starting-on-2019-02-28/22425), coinciding with this release, crates.io will require that you have a verified email address to publish. Starting at 2019-03-01 00:00 UTC, if you don't have a verified email address and run `cargo publish`, you'll get an error.

This ensures we can comply with DMCA procedures. If you haven't heeded the warnings cargo printed during the last release cycle, head on over to [crates.io/me](https://crates.io/me) to set and verify your email address. This email address will never be displayed publicly and will only be used for crates.io operations.

```